lib/step-value.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+const inherits      = require('util').inherits;
+const EventEmitter  = require('events').EventEmitter;
+
+class StepValue {
+  constructor(compiled, definitions) {
```

Comment:
You may want to add an argument default to `definitions`, so that we can guarantee `definitions.length` will return a `number`

so, `constructor(compiled, definitions = [])`. You could default compiled as well, but maybe throw an error if not?